chia/rpc/data_layer_rpc_api.py

```diff
@@ -201,12 +202,24 @@ async def batch_update(self, request: Dict[str, Any]) -> EndpointResult:
         fee = get_fee(self.service.config, request)
         changelist = [process_change(change) for change in request["changelist"]]
         store_id = bytes32(hexstr_to_bytes(request["id"]))
+        publish_on_chain = request.get("publish_on_chain", True)
         # todo input checks
         if self.service is None:
             raise Exception("Data layer not created")
-        transaction_record = await self.service.batch_update(store_id, changelist, uint64(fee))
-        if transaction_record is None:
-            raise Exception(f"Batch update failed for: {store_id}")
+        transaction_record = await self.service.batch_update(store_id, changelist, uint64(fee), publish_on_chain)
+        if publish_on_chain:
+            if transaction_record is None:
+                raise Exception(f"Batch update failed for: {store_id}")
+            return {"tx_id": transaction_record.name}
+        else:
+            if transaction_record is not None:
+                raise Exception("Transaction published on chain, but publish_on_chain set to False")
+            return {}
```

Comment:
Should we just always contain the `tx_id` key?  I don't know what we tend to do in our RPCs regarding this question.  There are at least a couple more cases below.